client/state/user-settings/reducer.js

```diff
@@ -32,26 +33,30 @@ export const settings = ( state = {}, { type, settingValues } ) => {
 
 export const unsavedSettings = ( state = {}, action ) => {
 	switch ( action.type ) {
-		case USER_SETTINGS_UNSAVED_CLEAR:
-			// After a successful save, remove the saved settings (either all of them,
-			// or a subset) from the `unsavedSettings`.
-			if ( ! action.settingNames ) {
+		case USER_SETTINGS_UNSAVED_CLEAR: {
+			const settingNames = action.settingNames;
+
+			if ( ! settingNames ) {
 				return {};
 			}
-			return omit( state, action.settingNames );
 
-		case USER_SETTINGS_UNSAVED_SET:
-			if ( state[ action.settingName ] === action.value ) {
-				return state;
+			if ( Array.isArray( settingNames ) ) {
+				return settingNames.reduce( removeValue, state );
 			}
-			return { ...state, [ action.settingName ]: action.value };
-
-		case USER_SETTINGS_UNSAVED_REMOVE:
-			return omit( state, action.settingName );
 
-		default:
-			return state;
+			return removeValue( state, settingNames );
+		}
+		case USER_SETTINGS_UNSAVED_SET: {
+			if ( get( state, action.settingName ) === action.value ) {
```

Comment:
Do we really need to use `get` here? Could we use `state?.[ action.settingName ]` instead?

Reply:
Yes, we need to be able to access a nested property like `calypso_preferences.colorScheme` which wouldn't work with `state?.[ action.settingName ]`. We may be able to introduce our own `getValue` helper here but that seems to be relevant to all of Calypso, I guess?

Reply:
Seems like that when we're introducing a `set` and `remove` value, it would be useful to have a corresponding `get`. Not necessarily in this PR, but definitely something we could use instead of lodash in the future.

Reply:
Yes, although I have to say that e.g. `removeValue` is somewhat special because it not only "unsets" a property but it also removes empty (sub-)objects.